packages/flutter/lib/src/cupertino/toggleable.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// A mixin for [StatefulWidget]s that implement iOS-themed toggleable
+/// controls (e.g.[Checkbox]es).
+///
+/// The mixin implements the logic for toggling the control when tapped.
+/// It does not have any opinion about the visual representation of the
+/// toggleable widget. The visuals are defined by a [CustomPainter] passed to
+/// the [buildToggleable]. [State] objects using this mixin should call that
+/// method from their [build] method.
+///
+/// This mixin is used to implement the Cupertino components for [Checkbox]
```

Comment:
This isn't too clear, because people will go there and see that `Checkbox` is in the Material components.  You might (briefly) explain how `Checkbox` and `CupertinoCheckbox` are related.